src/WheelsAndGoods.Application/Services/AuthService.cs

```diff
@@ -10,41 +13,65 @@ namespace WheelsAndGoods.Application.Services;
 public class AuthService : IAuthService
 {
 	private readonly IUnitOfWork _unitOfWork;
+	private readonly AuthOptions _authOptions;
 	private readonly IHashingProvider _hashingProvider;
 	private readonly IApplicationMapper _mapper;
 
-	public AuthService(IUnitOfWork unitOfWork, IHashingProvider hashingProvider, IApplicationMapper mapper)
+	public AuthService(IUnitOfWork unitOfWork, IHashingProvider hashingProvider, IApplicationMapper mapper, IOptions<AuthOptions> options)
 	{
 		_unitOfWork = unitOfWork;
 		_hashingProvider = hashingProvider;
 		_mapper = mapper;
+		_authOptions = options.Value;
 	}
 
-	public async Task<UserInfoResponse> CreateUser(RegisterRequest request)
+	public async Task<UserResponse> CreateUser(RegisterRequest request)
 	{
 		bool emailInUse = await _unitOfWork.Users.IsEmailExists(request.Email);
 
 		if (emailInUse)
 		{
 			throw new ApplicationException("Email is already in use");
 		}
-		
+
 		bool phoneInUse = await _unitOfWork.Users.IsPhoneExists(request.Phone);
 
 		if (phoneInUse)
 		{
 			throw new ApplicationException("Phone is already in use");
 		}
-        
+
 		User user = _mapper.ToUser(request, _hashingProvider);
-		
+
 		await _unitOfWork.CommitTransactionAsync(() =>
 		{
 			_unitOfWork.Users.Add(user);
 		});
-		
-		// TODO: Add creation JWT for user
-		
-		return _mapper.ToUserInfoResponse(user);
+
+		return _mapper.ToUserResponse(await GenerateAuthResponse(user), _mapper.ToUserInfoResponse(user));
```

Comment:
I will suggest moving GenerateAuthResponse to the top, it will be more readable. And also sub mapping should be moved into mapper.
```
var jwtResponse = GenerateAuthResponse(user);
return _mapper.ToUserResponse(jwtResponse, user);
```

Then update mapper to:
```
	public UserResponse ToUserResponse(JwtResponse jwtResponse, User user)
	{
		return new UserResponse()
		{
			Jwt = jwtResponse,
			User = ToUserInfoResponse(user),
		};
	}
```